components/nexus-core/src/main/java/org/sonatype/nexus/tasks/descriptors/ReconcileChecksumsTaskDescriptor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.tasks.descriptors;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.formfields.FormField;
+import org.sonatype.nexus.formfields.RepoOrGroupComboFormField;
+import org.sonatype.nexus.formfields.StringTextFormField;
+
+/**
+ * @since 2.11.3
+ */
+@Named("ReconcileChecksums")
+@Singleton
+public class ReconcileChecksumsTaskDescriptor
+    extends AbstractScheduledTaskDescriptor
+{
+  public static final String ID = "ReconcileChecksumsTask";
+
+  public static final String REPO_OR_GROUP_FIELD_ID = "repositoryId";
+
+  public static final String RESOURCE_STORE_PATH_FIELD_ID = "resourceStorePath";
+
+  private final RepoOrGroupComboFormField repoField = new RepoOrGroupComboFormField(REPO_OR_GROUP_FIELD_ID,
```

Comment:
If we allow a choice for narrowing the application, doesn't that leave us in danger of missing bad checksums? If this could affect any repository, might be better to remove the choice.